### ISSUE TITLE
Wrong city in addresses

### ISSUE THREAD

**Issue by rightsaidfred99** [organicmaps/organicmaps#857]
Hi there,

I'm.not sure if this is a problem with OpenMaps data or not but the capital city of Australia, Canberra - all the addresses appear as Sydney. 

So instead of "123 street, Suburb, Canberra ACT"
OrganicMaps displays "123 street, Suburb, Sydney"

This is a mistake and happens with all addresses, even the places to east suggestions.

**Comment by biodranik** [comment]
@vng can you please take a look?

**Comment by amo-f77** [comment]
I wonder if it's because Canberra is part of the Sydney, Australia map.

When I search for 2 yamba place, Canberra. I get a suggestion of
**`Yamba Place, 2`**
`Building`
`Canberra, Yamba Place, 2, Sydney, Australia`

It has the right information, with Sydney, Australia tacked on the end.

**Comment by rightsaidfred99** [comment]
> 
> 
> I wonder if it's because Canberra is part of the Sydney, Australia map.
> 
> When I search for 2 yamba place, Canberra. I get a suggestion of
> **`Yamba Place, 2`**
> `Building`
> `Canberra, Yamba Place, 2, Sydney, Australia`
> 
> It has the right information, with Sydney, Australia tacked on the end.

That's correct, I too think it's because it gets downloaded within the "Sydney" map. 

It should read:
Yambah Place, Narrabundah ACT, Australia (
[as per google maps](https://www.google.com.au/maps/place/Yamba+Pl,+Narrabundah+ACT+2604/@-35.3293333,149.1476882,17z/data=!3m1!4b1!4m5!3m4!1s0x6b164cf5f84b35f3:0xdcd227fec025f459!8m2!3d-35.3293333!4d149.1498769?hl=en)

It would be the same thing as saying Washington DC doesn't exist and should be included with New York map.  

1600 Pennsylvania Ave NW, New York, NY

**Comment by biodranik** [comment]
"Hawthorn" in Melbourne map area shows many search results with the same suburb name: "Melbourne, Australia"

**Comment by benji88i** [comment]
> "Hawthorn" in Melbourne map area shows many search results with the same suburb name: "Melbourne, Australia"

Yes this is all across Melbourne for many many suburbs, in fact it just says Melbourne everywhere. Where can we download the source code?

**Comment by biodranik** [comment]
> Where can we download the source code?

?

**Comment by benji88i** [comment]
Sorry, let me clarify. I meant source code for the addresses of each locality. Couldn't we take this from gnome maps or similar variant to correct this and adapt to suit for organic maps,

**Comment by biodranik** [comment]
No, this problem should be solved by properly checking administrative borders that are present in OSM.

**Comment by benji88i** [comment]
oh right, so how can I help? I know the city and the country well. How do I find the administrative borders?

**Comment by biodranik** [comment]
Please check this https://wiki.openstreetmap.org/wiki/Tag:boundary%3Dadministrative
And check if osm.org already has proper boundaries in your area.

**Comment by biodranik** [comment]
boundary=administrative admin_level=10 place=suburb are not supported yet in OM search results/addresses:

Searching for Birrabang Avenue in Organic Maps (NSW Australia) returns "Birrabang Avenue, Doyalson", but it should return "Birrabang Avenue, Summerland Point".

**Comment by rightsaidfred99** [comment]
> boundary=administrative admin_level=10 place=suburb are not supported yet in OM search results/addresses:
> 
> Searching for Birrabang Avenue in Organic Maps (NSW Australia) returns "Birrabang Avenue, Doyalson", but it should return "Birrabang Avenue, Summerland Point".

I don't understand, does this mean its not possible to fix on OrganicMaps' end? The OSM data needs to implement suburbs?

**Comment by biodranik** [comment]
It is possible to fix in OM but requires a non-trivial effort with borders. We'll see what we can do.

**Comment by biodranik** [comment]
I looked up a street, and it suggested it as being in 'Heerlen' the city next to it.

Street in question: Zeverij, Landgraaf, Limburg, Nederland
Noted by Organic Maps search: Zeverij, Landgraaf, Limburg, Nederland

I checked with OpenStreetMap (website), and that shows correctly 'Landgraaf'.

**Comment by biodranik** [comment]
> I encountered a bug while searching this particular street in Orléans ((Loiret, France) : rue Gustave Flaubert .
This street is located in Orléans la Source, with Zip code 45100.
In Osmand, the street is correctly linked with Orléans la Source but if you search it with Organic Maps, the street suggested is linked to Olivet ! If you select it, it is as a matter of fact the appropriate street but it would be problematic for someone searching it without actually knowing the place.
Curiously if you select the street on the map, it returns the right zip code.
Should you not know this city, Olivet is a city just beside Orléans and Orléans la Source is a part of Orléans with a specific zip code. Makes me think that some boundary between the cities is incorrect?

**Comment by LuisAPI** [comment]
Adding some 2c in here: https://fosstodon.org/@lastcelebration/109495492353526679

tl;dr: "it just says Melbourne everywhere" is also true of Greater Manila suburbs and exurbs

**Comment by Talldad** [comment]
I have experienced this also and have investigated Australia's Tagging Guidelines.

I recommend that OSM or OM (not sure where it needs to happen) be modified to respect the highest level of tagging in search results. In Australia, that would be 9 for locality/suburb.

**Comment by rightsaidfred99** [comment]
With the new dataset used in December 16, the bug for Australia sadly still exists

**Comment by biodranik** [comment]
> The street Am Breitenacker is located in search of Völklingen. However, it belongs to the independent municipality of Wadgassen, which is adjacent to Völklingen but has nothing to do with it.

**Comment by patepelo** [comment]
I got a report from a fellow OSM contributor noting the same problem. Here is the exmple: 
![image](https://github.com/organicmaps/organicmaps/assets/23003127/cc5360aa-05f6-4b15-846c-72eb96e42989)
![image](https://github.com/organicmaps/organicmaps/assets/23003127/03ada4c2-3698-4ab0-bef9-0ac8ab74842a)
![image](https://github.com/organicmaps/organicmaps/assets/23003127/a7af75a5-4c77-44f2-aa9e-e601bb3c5741)

In the example, "Táriba" is wrong City and the OSM data has following as "Junco Páramo": 
![image](https://github.com/organicmaps/organicmaps/assets/23003127/81f5b556-a23e-41d9-aa0d-846ed54b10c0)

**Comment by biodranik** [comment]
https://www.openstreetmap.org/node/7146640424

2 rue de Verlaine in Eybens displays 2 rue de Verlaine in St Martin d'Heres

![выява](https://github.com/organicmaps/organicmaps/assets/170263/96ff947a-b859-439f-85d5-ddf4971ee46b)

Another example:

https://www.openstreetmap.org/node/1938295622

2 Square Buffon in Eybens displays the city Scallops

**Comment by biodranik** [comment]
203 Apache Trail (39.94413, -74.54769) needs to be under Pemberton Township in the search results. Instead, it shows Manchester Township.

https://www.openstreetmap.org/way/641699162

**Comment by biodranik** [comment]
https://github.com/organicmaps/organicmaps/issues/4359

**Comment by biodranik** [comment]
A few more examples of wrong addresses:

![выява](https://github.com/user-attachments/assets/83c4957b-a42b-4f87-b562-e0292a6b07d1)
![выява](https://github.com/user-attachments/assets/f9afacf8-a498-45be-9e00-ec2d18c4b5de)
![выява](https://github.com/user-attachments/assets/a5f122c5-49fb-47ff-b466-25aed7651115)

**Comment by Niavy64** [comment]
Previous screenshots are mine. These are French address queries, results return Pau (64000) or Pessac (33700). Chemin du Saüt is in Monein (zipcode 64360) Allée du bois du stade is in Merignac (same zip code 33700). Allée Jupiter is in Lons (zip code 64140

**Comment by maxstarkenburg** [comment]
@rightsaidfred99's comment about DC seems to have been prescient. When I look up addresses in Washington, DC, they are listed as if they were in Maryland. 😒

![Screenshot_20241125-212457.png](https://github.com/user-attachments/assets/746bf7cb-1f3f-408f-88f7-0f9287f91c17)

I'm not seeing the same behavior if I search for the same addresses on openstreetmap.org.

**Comment by biodranik** [comment]
> Rue de l’Espérance in Créteil, Rue de l’Averse, and Rue Chéret — all the streets in this district are marked as being in Saint-Maur-des-Fossés in your application, although they are actually in Créteil.

> On OpenStreetMap and in other applications using these maps, these streets are properly shown as being in Créteil.

**Comment by vng** [comment]
One more example (for the issues search): "Belgrave" and "Woori Yallock".

**Comment by LaurentBerger** [comment]
[Rue de Rome is in La chapelle Saint aubin](https://www.openstreetmap.org/way/94552643/history#map=16/48.03157/0.16131) in openstrretmap 
but in Le Mans in organics Maps (2026.04.07-8-FDroid)

<img width="276" height="367" alt="Image" src="https://github.com/user-attachments/assets/ffdecf67-7966-4bb2-9356-00b15afdffb6" />
